src/care/RAJAPlugin.cpp

```diff
@@ -1,274 +0,0 @@
-   unsigned int RAJAPlugin::s_current_color = 0;
-
-   std::string RAJAPlugin::s_current_loop_file_name = "N/A";
-   int RAJAPlugin::s_current_loop_line_number = -1;
-
-   std::vector<const chai::PointerRecord*> RAJAPlugin::s_active_pointers_in_loop = std::vector<const chai::PointerRecord*>{};
-   std::unordered_map<void *, std::function<void(chai::ExecutionSpace, const char *, int)>> RAJAPlugin::s_post_parallel_forall_actions = std::unordered_map<void *, std::function<void(chai::ExecutionSpace, const char *, int)>>{};
-   int RAJAPlugin::s_threadID = -1;
-   /////////////////////////////////////////////////////////////////////////////////
-   ///
-   /// @brief Set up to be done before executing a RAJA loop.
-   ///
-   /// @arg[in] space The execution space
-   /// @arg[in] name The name of the loop
-   ///
-   /////////////////////////////////////////////////////////////////////////////////
-   void RAJAPlugin::pre_forall_hook(chai::ExecutionSpace space, const char* fileName, int lineNumber) {
-#if !defined(CHAI_DISABLE_RM)
-      // Update the CHAI execution space
-      if (s_update_chai_execution_space) {
-         chai::ArrayManager* arrayManager = chai::ArrayManager::getInstance();
-         arrayManager->setExecutionSpace(space);
-      }
-
-      // Prepare to record CHAI data
-      if (CHAICallback::isActive()) {
-         s_current_loop_file_name = fileName;
-         s_current_loop_line_number = lineNumber;
-         s_active_pointers_in_loop.clear();
-
-#if defined(CARE_GPUCC) && defined(CARE_DEBUG)
-         GPUWatchpoint::setOrCheckWatchpoint<int>();
-#endif // defined(CARE_GPUCC) && defined(CARE_DEBUG)
-      }
-#endif // !defined(CHAI_DISABLE_RM)
-
-#if defined(CARE_GPUCC)
-      // Synchronize
-      if (s_synchronize_before) {
-         if (space == chai::GPU) {
-            care::gpuDeviceSynchronize(fileName, lineNumber);
-         }
-      }
-
-#if CARE_HAVE_NVTOOLSEXT
-      // Profile the host loops
-      if (s_profile_host_loops) {
-         if (space == chai::CPU) {
-            std::string name = fileName + std::to_string(lineNumber);
-
-            int color_id = s_current_color++;
-            color_id = color_id % s_num_colors;
-
-            // TODO: Add error checking
-            nvtxEventAttributes_t eventAttrib = { 0 };
-            eventAttrib.version = NVTX_VERSION;
-            eventAttrib.size = NVTX_EVENT_ATTRIB_STRUCT_SIZE;
-            eventAttrib.colorType = NVTX_COLOR_ARGB;
-            eventAttrib.color = s_colors[color_id];
-            eventAttrib.messageType = NVTX_MESSAGE_TYPE_ASCII;
-            eventAttrib.message.ascii = name.c_str();
-            nvtxRangePushEx(&eventAttrib);
-         }
-      }
-#endif // CARE_HAVE_NVTOOLSEXT
-#endif // defined(CARE_GPUCC)
-   }
-
-   /////////////////////////////////////////////////////////////////////////////////
-   ///
-   /// @brief Writes out debugging information after a loop is executed.
```

Comment:
This doxygen style comment should be moved to the declaration of writeLoopData in DebugPlugin.h